### PR TITLE
build: enable clang-tidy by default (#4821)

### PR DESCRIPTION
**Pull request by rsekman**
In #4824 running clang-tidy was disabled by default due to an upstream bug that led to an infinite loop. See #4821 This bug has since been fixed in clang-tidy 18. It is therefore safe to re-enable clang-tidy by default. I have checked that the build completes on clang 18, version information below.

```
% uname -a
Linux naoetsu 6.9.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 28 Jun 2024 04:32:50 +0000 x86_64 GNU/Linux
% clang -v
clang version 18.1.8
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-pc-linux-gnu/13.3.0
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-pc-linux-gnu/14.1.1
Found candidate GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/13.3.0
Found candidate GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/14.1.1
Selected GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/14.1.1
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Selected multilib: .;@m64
Found CUDA installation: /opt/cuda, version 
% clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 18.1.8
  Optimized build.
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -71,7 +71,7 @@ option(ENABLE_NLS "Enable native language support" ON)
 option(INSTALL_DOC "Build/install documentation" ON)
 option(INSTALL_LIB "Install the library" OFF)
 tr_auto_option(ENABLE_DEPRECATED "Allow deprecated API use of upstream packages, e.g. GTK" AUTO)
-tr_auto_option(RUN_CLANG_TIDY "Run clang-tidy on the code" OFF)
+tr_auto_option(RUN_CLANG_TIDY "Run clang-tidy on the code" ON)
```
Comment (tearfur) [inline CMakeLists.txt:74]:
```suggestion
tr_auto_option(RUN_CLANG_TIDY "Run clang-tidy on the code" AUTO)
```

**Review by tearfur — CHANGES_REQUESTED** [review]

**Review by tearfur — APPROVED** [review]

**Review by mikedld — APPROVED** [review]

**Comment by mikedld** [comment]
Android build seems to find/use clang-tidy now, based purely on increased build times and it timing out. Let's try disabling it since we're not interested in it (?), probably somewhere here: https://github.com/transmission/transmission/blob/35ef8cf7a060bd5631ee0568ebd8a264480c650a/android/build.gradle#L35

**Review by ckerr — APPROVED** [review]

**Review by ckerr — CHANGES_REQUESTED** [review]
Seems to still be used on Android by default?

GitHub Actions has been going on the Android build for over four hours now. We can't merge something that causes CI to run for four hours for every change :skull:

**Comment by mikedld** [comment]
Seems like disabling clang-tidy for the Android CI build helped. I'm going to assume this satisfies both me and @ckerr and merge this.

**Pull request merged**